app/livechat/server/lib/Helper.js

```diff
@@ -0,0 +1,230 @@
+import { Meteor } from 'meteor/meteor';
+import { Match, check } from 'meteor/check';
+import { MongoInternals } from 'meteor/mongo';
+
+import { Messages, Rooms, Subscriptions, Users } from '../../../models';
+import { LivechatInquiry } from '../../lib/LivechatInquiry';
+import { Livechat } from './Livechat';
+import { RoutingManager } from './RoutingManager';
+
+export const createLivechatRoom = (rid, name, guest, extraData) => {
+	check(rid, String);
+	check(name, String);
+	check(guest, Match.ObjectIncluding({
+		_id: String,
+		username: String,
+		status: Match.Maybe(String),
+		department: Match.Maybe(String),
+	}));
+
+	const { _id, username, token, department: departmentId, status = 'online' } = guest;
+
+	const room = Object.assign({
+		_id: rid,
+		msgs: 0,
+		usersCount: 1,
+		lm: new Date(),
+		fname: name,
+		t: 'l',
+		ts: new Date(),
+		departmentId,
+		v: {
+			_id,
+			username,
+			token,
+			status,
+		},
+		cl: false,
+		open: true,
+		waitingResponse: true,
+	}, extraData);
```

Comment:
what do you think about: `{ "more fields here", ...extraData }`

Reply:
Yeah, I was doing it here: https://github.com/RocketChat/Rocket.Chat/pull/15003#discussion_r313097289
:)